controllers/machine_controller.go

```diff
@@ -199,6 +196,22 @@ func (r *MachineReconciler) patchStatus(ctx context.Context, bm *bmcv1alpha1.Mac
 	return ctrl.Result{}, nil
 }
 
+// convertRawBMCPowerState takes a raw BMC power state response and attempts to convert it to
+// a PowerState.
+func convertRawBMCPowerState(response string) (v1alpha1.PowerState, error) {
```

Comment:
Would you mind adding a test for this function? Also, would you be open to a rename suggestion here? maybe `toPowerState` or `rawToPowerState`?

Reply:
It's unexported so is it safe to assume you mean white box tests? I usually subscribe to the _test package idiom unless an unexported func is necessarily complex. What're other projects doing?

Fwiw I have a branch where I'm working on a suite of black box unit tests that'll ultimately resolve #30.

Reply:
#55 is now up. We can merge that prior to this and then ensure things are covered adequately.